crates/sage/src/scoring.rs

```diff
@@ -156,12 +163,45 @@ fn lnfact(n: u16) -> f64 {
     }
 }
 
+fn log_factorial(n: u16, k: u16) -> f64 {
```

Comment:
Why use this instead of the ln_fact function? If I wanted to calculate a precise log factorial instead of an approximation, I would already be doing so :)

Reply:
I wanted to stay true to the original implementation, but it should be equivalent to using the `ln_fact` function with approximation instead.

Reply:
If there's a big difference, we can generate a pre-computed lookup table and use that.